### PR TITLE
feat: Support padding tokens with seqlen=0 for rope+quant+kv cache update fusion kernel

### PR DESCRIPTION
**Pull request by elvischenv**
## 📌 Description

vLLM is using seqlen=0 padding tokens for running a full cudagraph: https://github.com/vllm-project/vllm/blob/95c0f928cdeeaa21c4906e73cee6a156e1b3b995/vllm/v1/worker/gpu/model_runner.py#L652-L654

Update the following functions:
`get_batch_indices_positions_kernel`: initialize `batch_indices`/`positions` to `-1`/`0` for recognizing the padding tokens
`rope_quantize_fp8_append_paged_kv_cache`: skip those padding tokens

Testing:
`pytest -v -s tests/attention/test_rope.py::test_rope_quantize_fp8_append_paged_kv_cache_padding`
```
======= 4 passed in 1.66s =======
```

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [x] All tests are passing (`unittest`, etc.).

## Summary by CodeRabbit

* **Bug Fixes**
  * Paged KV cache append now reliably skips padding entries, preventing unintended overwrites during attention quantization and positional encoding processing.
  * Deterministic initialization of internal index/position buffers to avoid uninitialized memory affecting padding behavior.

* **Tests**
  * Added parameterized regression test confirming padding entries are ignored and prefilled KV cache data remains exactly unchanged.

### CODE REVIEW

File: include/flashinfer/pos_enc.cuh [hunk 1]
```diff
@@ -859,169 +859,175 @@ __global__ void RopeQuantizeAppendPagedKVCacheKernel(
...
-    // Load cos/sin for RoPE processing blocks only
-    if ((tx * vec_size < rope_dim) && (by < k_rope_end)) {
-      int sin_offset = rope_dim / 2;
-      int vec_idx;
-      if constexpr (interleave) {
-        vec_idx = (tx * vec_size) / 2;  // Force integer division
-      } else {
-        vec_idx = (tx * vec_size) % half_rope_dim;
+    // skip padding tokens with batch_indices < 0
+    if (batch_indices[idx] >= 0) {
```
Comment (elvischenv) [inline include/flashinfer/pos_enc.cuh:863]: The main change is just this line. The following is just indent formatting.

**Comment by elvischenv** [comment]
Hi @yzh119, could you help review this? We need this fix for integrating this kernel to vLLM. Thanks!

**Comment by elvischenv** [comment]
cc @kahyunnam for viz.

**Comment by nvpohanh** [comment]
@elvischenv could you rebase again?

**Comment by nvpohanh** [comment]
@bkryu Could you review this or assign someone to review this? Thanks!

**Comment by nvpohanh** [comment]
@bkryu could you review this? Thanks

File: flashinfer/page.py
Comment (bkryu) [inline flashinfer/page.py:1]: Hi @elvischenv, the changes generally look correct and the CI does seem to pass.

However, Have you tried measuring the performance implications? Asking because `torch.full` or `torch.zeros` tend to call memset which have a higher overhead than `torch.empty`. I'm wondering whether there will be a noticeable performance difference from it.

Comment (nvpohanh) [reply flashinfer/page.py:1]: @elvischenv is this called per decoding step or per attention layer? If this is per decoding step, I am less worried about the additional memsets. But if this is per-layer, it may be noticeable perf overhead.

Comment (elvischenv) [reply flashinfer/page.py:1]: `get_batch_indices_positions` is a helper function, preparing the needed arguments for `rope_quantize_fp8_append_paged_kv_cache`, should only be called per decoding step. Then the whole iteration can reuse the same `batch_indices` and `positions`, which won't produce the noticeable overhead.

Comment (nvpohanh) [reply flashinfer/page.py:1]: @bkryu Once per decoding step should be okay? Do you agree?

**Comment by nvpohanh** [comment]
@bkryu Are the failures known ones or caused by this PR?

File: flashinfer/page.py
Comment (bkryu) [reply flashinfer/page.py:1]: I agree that it should be fine.

**Review by bkryu — APPROVED** [review]
CI failures are unrelated. LGTM!
